examples/private_set_intersection/module/rust/src/lib.rs

```diff
@@ -49,32 +49,46 @@ oak::entrypoint!(oak_main => |_in_channel| {
 pub const SET_THRESHOLD: u64 = 2;
 
 #[derive(Default)]
-struct Node {
+struct SetIntersection {
     /// Current intersection of contributed private sets.
-    values: Option<HashSet<String>>,
+    values: HashSet<String>,
     /// Number of contributed private sets.
     set_count: u64,
     /// The intersection is locked and new contributions are discarded.
     locked: bool,
 }
 
+#[derive(Default)]
+struct Node {
+    /// Map from set ID to `SetIntersection`.
```

Comment:
Could you explain a bit more here? e.g. that this allows multiple sets of clients to compute their own intersections, also explain what the security characteristics are, in particular I think currently just specifying a different set id does not guarantee that data from different clients is kept separate, right? That would require some more sophisticated labelling.